### PR TITLE
Remove `serde` requirement from FRAME macros

### PR DESCRIPTION
**Pull request by bkchr**
Currently there is some implicit requirement on `serde` being present in
the `Cargo.toml` of a pallet when `GenesisConfig` is used. This pr
removes this requirement by using the serde attribute `serde(crate = "..")`.

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]

File: primitives/runtime/src/traits.rs [hunk 1]
```diff
@@ -1217,6 +1217,8 @@ macro_rules! impl_opaque_keys {
 			)*
 		}
 	) => {
+		#[cfg(feature = "std")]
+		use $crate::serde as __opaque_keys_serde_import__;
```
Comment (thiolliere) [inline primitives/runtime/src/traits.rs:1221]: the polkadot build complains: the name `__opaque_keys_serde_import__` is defined multiple times

But if we abuse from paste featureto concatenate string in doc, we can do:

```diff
diff --git a/primitives/runtime/src/traits.rs b/primitives/runtime/src/traits.rs
index f95a7aa200..bdd5efde59 100644
--- a/primitives/runtime/src/traits.rs
+++ b/primitives/runtime/src/traits.rs
@@ -1216,9 +1216,36 @@ macro_rules! impl_opaque_keys {
                                pub $field:ident: $type:ty,
                        )*
                }
+       ) => {
+               $crate::paste::paste!(
+                       $crate::impl_opaque_keys!(@with_serde_crate_literal
+                               $( #[ $attr ] )*
+                               pub struct $name {
+                                       $(
+                                               $( #[ $inner_attr ] )*
+                                               pub $field: $type,
+                                       )*
+                               }
+                               // NOTE: we abuse from `paste` feature which concatenate string in docs to build
+                               // our string
+                               #[doc = "__opaque_keys_serde_import__" $name ""]
+                       );
+               );
+       };
+       (@with_serde_crate_literal
+               $( #[ $attr:meta ] )*
+               pub struct $name:ident {
+                       $(
+                               $( #[ $inner_attr:meta ] )*
+                               pub $field:ident: $type:ty,
+                       )*
+               }
+               #[ doc = $serde_crate_literal:expr ]
        ) => {
                #[cfg(feature = "std")]
-               use $crate::serde as __opaque_keys_serde_import__;
+               $crate::paste::paste!(
+                       use $crate::serde as [< __opaque_keys_serde_import__ $name >];
+               );
                $( #[ $attr ] )*
                #[derive(
                        Default, Clone, PartialEq, Eq,
@@ -1227,7 +1254,7 @@ macro_rules! impl_opaque_keys {
                        $crate::RuntimeDebug,
                )]
                #[cfg_attr(feature = "std", derive($crate::serde::Serialize, $crate::serde::Deserialize))]
-               #[cfg_attr(feature = "std", serde(crate = "__opaque_keys_serde_import__"))]
+               #[cfg_attr(feature = "std", serde(crate = $serde_crate_literal))]
                pub struct $name {
                        $(
                                $( #[ $inner_attr ] )*
```

Comment (bkchr) [reply primitives/runtime/src/traits.rs]: Ty ❤️

Comment (bkchr) [reply primitives/runtime/src/traits.rs]: I also thought about something similar 😬

File: primitives/runtime/src/traits.rs [hunk 2]
```diff
@@ -1216,9 +1216,36 @@ macro_rules! impl_opaque_keys {
...
+				$( #[ $attr ] )*
+				pub struct $name {
+					$(
+						$( #[ $inner_attr ] )*
+						pub $field: $type,
+					)*
+				}
+				// NOTE: we abuse from `paste` feature which concatenate string in docs to build
+				// our string
+				#[doc = "__opaque_keys_serde_import__" $name]
```
Comment (bkchr) [inline primitives/runtime/src/traits.rs:1231]: Ahh but wait. We don't need to do this soo complicated

Comment (thiolliere) [reply primitives/runtime/src/traits.rs]: ah great

**Pull request merged**